lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp

```diff
@@ -133,11 +133,33 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
 
   auto instrumentableFunctions = PIXPassHelpers::GetAllInstrumentableFunctions(*m_DM);
 
-  for (auto * F : instrumentableFunctions) {
+  for (auto *F : instrumentableFunctions) {
+    for (auto &block : F->getBasicBlockList()) {
+      for (llvm::Instruction &I : block.getInstList()) {
+        AnnotateValues(&I);
+      }
+    }
+  }
+
+  for (auto *F : instrumentableFunctions) {
+    for (auto &block : F->getBasicBlockList()) {
+      for (llvm::Instruction &I : block.getInstList()) {
+        AnnotateStore(&I);
```

Comment:
It looks as though AnnotateStore might depend on the metadata set by AnnotateValues above. Is that correct? Given how egregious the repeated looping looks, it might be worth mentioning in a comment if that's the case.

Even if it is the case, I think it's unlikely that there is ever a store with arguments that haven't been annotated even with one loop. I don't know if I'd risk it without more certainty than I have, but since this change is meant to speed things up, this might be another way to help there.

Reply:
The performance concern is around compiling and running the instrumented shader (driver compile and runtime performance), rather than the performance of this pass.

Reply:
Ach I went and completed the PR before noticing this comment. Darn github timeout. 

Yes, what Tex said, but yes a comment is a good idea. I'll be back in amongst this code soon, I think (for typedefs), so I'll fix it up then.